homeassistant/components/bang_olufsen/event.py

```diff
@@ -0,0 +1,89 @@
+"""Event entities for the Bang & Olufsen integration."""
+
+from __future__ import annotations
+
+from homeassistant.components.event import EventDeviceClass, EventEntity
+from homeassistant.const import CONF_MODEL
+from homeassistant.core import HomeAssistant, callback
+from homeassistant.helpers.dispatcher import async_dispatcher_connect
+from homeassistant.helpers.entity_platform import AddEntitiesCallback
+
+from . import BangOlufsenConfigEntry
+from .const import (
+    CONNECTION_STATUS,
+    DEVICE_BUTTON_EVENTS,
+    DEVICE_BUTTONS,
+    MODEL_SUPPORT_DEVICE_BUTTONS,
+    MODEL_SUPPORT_MAP,
+    WebsocketNotification,
+)
+from .entity import BangOlufsenEntity
+
+
+async def async_setup_entry(
+    hass: HomeAssistant,
+    config_entry: BangOlufsenConfigEntry,
+    async_add_entities: AddEntitiesCallback,
+) -> None:
+    """Set up Sensor entities from config entry."""
+
+    entities: list[EventEntity] = []
+
+    # Add physical "buttons"
+    if config_entry.data[CONF_MODEL] in MODEL_SUPPORT_MAP[MODEL_SUPPORT_DEVICE_BUTTONS]:
+        entities.extend(
+            [
+                BangOlufsenButtonEvent(config_entry, button_type)
+                for button_type in DEVICE_BUTTONS
+            ]
+        )
+
+    async_add_entities(new_entities=entities)
+
+
+class BangOlufsenEvent(BangOlufsenEntity, EventEntity):
+    """Base Event class."""
+
+    _attr_entity_registry_enabled_default = False
+
+    @callback
+    def _async_handle_event(self, event: str) -> None:
+        """Handle event."""
+        self._trigger_event(event)
+        self.async_write_ha_state()
+
+
+class BangOlufsenButtonEvent(BangOlufsenEvent):
+    """Event class for Button events."""
+
+    _attr_device_class = EventDeviceClass.BUTTON
+    _attr_event_types = DEVICE_BUTTON_EVENTS
+    _attr_icon = "mdi:gesture-tap-button"
```

Comment:
Instead of hardcoding icons, icon translations should be used. https://developers.home-assistant.io/blog/2024/01/19/icon-translations/

Reply:
Done.